### PR TITLE
Added Instance method tolist() and Frontend test

### LINKED ISSUE #11758: tolist

### PR DESCRIPTION
**Pull request by cressidasuphina**
Close #11758

### CODE REVIEW

**Comment by cressidasuphina** [comment]
Hi, could you help me? My values ret_np_flat and ret_from_np_gt_flat don't match. But I compared it manually and it looks the same. As I got it right ground truth framework is original Numpy library itself, isn't it? Also, I'm not sure if it's ok that I added ivy.to_list(x) to manipulation_routines/changing_kind_of_array.py file. 
image
image

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @cressidasuphina 

Would you add the unit tests for `to_list` function as well and also remove the changes from `.idea` files 

Thanks

**Comment by cressidasuphina** [comment]
Hi @hirwa-nshuti thank you for your comment, I did what you asked but unfortunately I have got an an exception in test_frontend_function
image

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @cressidasuphina 

The reason why your tests are failing is because `numpy` doesn't have `tolist` as a function it is only and instance method of `np.ndarray` https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tolist.html

You need to implement it only in the `ndarray ` class methods

Thanks

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @cressidasuphina 

This function isn't yet implemented on the numpy frontend.
You'll have to implement its composition😊

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -291,6 +291,9 @@ def std(
     def tobytes(self, order="C") -> bytes:
         return np_frontend.tobytes(self.data, order=order)
 
+    def tolist(self) -> list:
+        return np_frontend.ivy.to_list(self._ivy_array)
```
[the remark this answers is not in the document]
Comment (cressidasuphina) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py]: Hello, thank you for your comment and your attention. I hope I have got it right this time.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @cressidasuphina 

There is some progress now😊 However the function `tolist` is not present in the numpy frontend namespace, you'll have to implement it using compositions

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -2563,7 +2563,38 @@ def test_numpy_instance_tobytes__(
...
+        init_all_as_kwargs_np={
+            "object": x[0],
+        },
+        method_input_dtypes=input_dtypes,
+        method_all_as_kwargs_np={},
+        init_flags=init_flags,
+        method_flags=method_flags,
+        frontend=frontend,
+        frontend_method_data=frontend_method_data,
+        on_device=on_device,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2595]:
```suggestion
        on_device=on_device,
        test_values=False, #Todo change this after we add __iter__ to ndarray
```

**Review by hirwa-nshuti — APPROVED** [review]
Hey @cressidasuphina 

Thanks for the PR
All looks good going to merge😊

**Comment by cressidasuphina** [comment]
Hi

**Pull request merged**